cmd/queue/main.go

```diff
@@ -91,6 +95,12 @@ func initEnv() {
 
 	// TODO(mattmoor): Move this key to be in terms of the KPA.
 	servingRevisionKey = autoscaler.NewKpaKey(servingNamespace, servingRevision)
+	health = &healthServer{alive: true}
+	_reporter, err := queue.NewStatsReporter(servingNamespace, servingConfig, servingRevision)
+	if err != nil {
+		logger.Fatal("Failed to create stats reporter", zap.Error(err))
+	}
+	reporter = _reporter
```

Comment:
Should this be moved down into main, closer to the initialization of the exporter endpoint, to have it all packed in one place?

I think the `initEnv` function is supposed to read environment values into globals.

Reply:
It needs to be initialized before the health exporter is used due to the dependency it has on reporting now. And I see no indication on what `initEnv` should be used for by this I mean I can initialization happening in there that is not related to environment values...